### DISCUSSION TITLE
NextJs getRequestHandler context

### DISCUSSION THREAD

**Discussion by stychu** [vercel/next.js#61445] [Ideas]
### Goals

1. Close the gap between custom server and nextjs
2. Easily share data from custom server and nextjs handle

### Background

From what I found at the moment of time being it's not possible to easily share data between custom server (like expressjs) and nextjs. I have a custom express server nextjs app (app router 14.x) where my session is handled on the express side which has access to the user session from req/res object and in order to share that session I need a hacky solution with AsyncLocalStorage to access that user session data in nextjs land (api, rsc, etc.) 
I've seen a lot of folks having custom server and the same problem and doing hacky solutions to solve it... So It seems a valid proposal

### Proposal

Add a `context` to the `getRequestHandler` which will be available in the nextjs api routes, middleware etc to easily share and access data from the custom server.

```ts
import next from 'next';

const app = next({
  dev: serverEnv.NODE_ENV !== 'production',
  port: serverEnv.PORT,
});
const handle = app.getRequestHandler();

...
  server.all('*', (req: Request, res: Response, next: NextFunction) => {
	  // HERE we could build the context object and pass it to the handle to be available in nextjs
      handle(req, res).catch(next);
    });

```

Heres how Remix handles that you can create the context which is available across remix API

```ts
const {
  createRequestHandler,
} = require("@remix-run/express");
const express = require("express");

const app = express();

// needs to handle all verbs (GET, POST, etc.)
app.all(
  "*",
  createRequestHandler({
		....
    // return anything you want here to be available as `context` in your
    // loaders and actions. This is where you can bridge the gap between Remix
    // and your server
    getLoadContext(req, res) {
      return {};
    },
  })
);

```

**Comment by xino1010** [comment]
what about this?
